library/src/blas3/rocblas_trmm.hpp

```diff
@@ -0,0 +1,1488 @@
+        rocblas_int ldt1 = rb, ldt2 = cb;
+        T*          dt1 = workspace;
+        T*          dt2 = workspace + rb * cb;
+
+        rocblas_int    offd = rocblas_diagonal_unit == diag ? 1 : 0;
+        rocblas_int    isec, jsec, tsec;
+        T              delta;
+        rocblas_status status = rocblas_status_success;
+
+        if(side == rocblas_side_left)
+        {
+            if(uplo == rocblas_fill_upper)
+            {
+                if(transa == rocblas_operation_none)
+                {
+                    //
+                    //              Form  C := alpha*A*C. Left, Upper, No transpose.
+                    //
+                    delta     = *alpha;
+                    bool cldc = dcld(ldc);
+                    for(int ii = 1; ii <= m; ii += cb)
+                    {
+                        isec = cb < m - ii + 1 ? cb : m - ii + 1;
+                        //
+                        //                  T2 := A', the transpose of a upper unit or non-unit
+                        //                  triangular diagonal block of A is copied to the
+                        //                  lower triangular part of T2.
+                        //
+                        for(int i = ii + offd; i <= ii + isec - 1; i++)
+                        {
+                            CHECK_INTERNAL_ROCBLAS_ERROR(
+                                (rocblas_copy_template<NB, T>(handle,
+                                                              i - ii + 1 - offd,
+                                                              &a[ii - 1 + (i - 1) * lda],
+                                                              0,
+                                                              1,
+                                                              0,
+                                                              &dt2[i - ii],
+                                                              0,
+                                                              cb,
+                                                              0,
+                                                              1)));
+                        }
+                        for(int jj = 1; jj <= n; jj += rb)
+                        {
+                            jsec = rb < n - jj + 1 ? rb : n - jj + 1;
+                            //
+                            //                      T1 := C', the transpose of a rectangular block
+                            //                      of C is copied to T1.
+                            //
+                            if(cldc)
+                            {
+                                for(int j = jj; j <= jj + jsec - 1; j++)
+                                {
+                                    CHECK_INTERNAL_ROCBLAS_ERROR(
+                                        (rocblas_copy_template<NB, T>(handle,
+                                                                      isec,
```

Comment:
Same as above

Reply:
T is required, cannot be deduced, moved parenthesis to be around function name and templates.